server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -308,6 +308,23 @@ Runnable getGlobalCheckpointSyncer() {
     private final Store remoteStore;
     private final TranslogFactory translogFactory;
 
+    public boolean isBlockInternalCheckPointRefresh() {
+        return blockInternalCheckPointRefresh;
+    }
+
+    public void setBlockInternalCheckPointRefresh(boolean blockInternalCheckPointRefresh) {
+        this.blockInternalCheckPointRefresh = blockInternalCheckPointRefresh;
+    }
+
+    /**
+     * Used with segment replication only.
+     *
+     * The flag is meant to block the segment replication events to all replicas from existing primary. This is done in
+     * order to ensure that file copied during peer recovery does not conflict with segment files from new primary post
+     * peer recovery.
+     */
+    private boolean blockInternalCheckPointRefresh;
```

Comment:
During a primary relocation handoff all operation to the older primary is drained to a queue which means there will be no operations indexed into older primary, at this point can we perform a final commit/refresh, wondering if we need to explicitly block refreshes at this point since refreshes wouldn't create segments

Reply:
Thank you @Bukhtawar for suggesting this. Yes, it makes sense to delay the segment replication round on primary target till relocation handoff in order to avoid blocking refresh. Let me try this approach.

Reply:
We need to just make sure it doesn't take too long else operations would pile up and the indexing operation permits blocked for longer

Reply:
I updated the logic and it seems to work fine (integ tests inside SegmentReplicationRelocationIT passes). I suspect the force segment replication shouldn't take much time as target is already synced during phase 1. I am planning to quantify the lag added by Segment replication with different shard/segment sizes on a real cluster. I will update with results here.

Reply:
@Bukhtawar @mch2  : Tried the changes on local machine and I see seg rep took additional ~3.5 secs during relocation handoff. https://github.com/opensearch-project/OpenSearch/issues/5242#issuecomment-1368133952

Reply:
> With our current impl I believe any ongoing event will simply cancel and start over from new primary.

@mch2 : I think this is not currently handled as peer recovery doesn't make primary term bump. I wrote a integ test which detects this [here](https://github.com/opensearch-project/OpenSearch/commit/647f6b860c40c17398ad3bc523ed25d5f59adf62). It was brought up previously by @ashking94 in [thread above](https://github.com/opensearch-project/OpenSearch/pull/5344#discussion_r1042555829)